end_to_end/end_to_end_suite_test.go

```diff
@@ -2511,4 +2512,61 @@ LANGUAGE plpgsql NO SQL;`)
 			}
 		})
 	})
+	Describe("Exclude subpartitions for given root partition in leaf-partition-data mode", func() {
```

Comment:
LGTM. Several small questions: why you prefer this test suite, not `filtered_test.go`? Maybe test restore from backups to test the problem described in analytical note?

Reply:
> why you prefer this test suite, not `filtered_test.go`?

It seemed to me that this is a suitable place for tests, but I can move it.

Reply:
> Maybe test restore from backups to test the problem described in analytical note?

This can be added.

Reply:
I moved the tests to filtered_test.go and rewrote them using restore checking.